### PR TITLE
GP11 : DSA-SHA224 DSA-SHA256 algorithms

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**

### CODE REVIEW

**Comment by cedric-chaumont-st-dev** [comment]
For testing, use optee_test PR#26

File: core/tee/tee_svc_cryp.c [hunk 1]
```diff
@@ -3420,7 +3422,12 @@ TEE_Result tee_svc_asymm_verify(uint32_t state,
 	if (res != TEE_SUCCESS)
 		goto out;
 
-	if (data_len != hash_size) {
+    	/*
+    	 * Depending on the DSA algorithm, the size of the hash to consider
+	 * could be less or equal than he whole size of the
+	 * hash algorithm involved (related to the subprime).
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: You mean the hash can be truncated when signed?

Comment (cedric-chaumont-st-dev) [inline core/tee/tee_svc_cryp.c]: Yes. For DSA-SH256, the hash value generated is coded on 256 bits, but the DSA  (encrypted hash value) could be either on 224bits or 256 bits (depending on the subprime).

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: core/tee/tee_svc_cryp.c (hunk 1 above)
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Is the subprime something like the modulus in RSA, that is, defining the size of the result of the operation?

Please update the comment with that clarification, also a short note on why it does not affect the security of the other algorithms would nice.

**Comment by cedric-chaumont-st-dev** [comment]
Updated. Squashed. Verified.
+-----------------------------------------------------
37980 subtests of which 1 failed
687 test cases of which 1 failed
0 test case was skipped
TEE test application done!

NOTE: GP test 7671 failing (patch on going)

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
Please rebase on master and add the tags

**Comment by cedric-chaumont-st-dev** [comment]
Rebase. Update git comment.

**Comment by pascal-brand-st** [comment]
Merged manually

**Pull request closed without merging**